### PR TITLE
Add transaction modal mostly completed

### PR DESCRIPTION
**Pull request by lionelgwk**

### CODE REVIEW

File: components/AddTransactionModal.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            // console.log(logos);
+        } catch (error) {
+            console.error("Error:", error);
+        }
+    };
+
+    // const fetchLogo = async (ticker: string) => {
+    //     try {
+    //         // const ticker = "AAPL";
+    //         const apiKey = "7S5YGbLiYc2AyLf78JKrsg==XjsnlMf8Lp9Yw7L9"; // Replace with your actual API key
```
Comment (DavidLHW) [inline components/AddTransactionModal.tsx:110]: exposed api key

File: components/AddTransactionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                                />
+                            </PopoverContent>
+                        </Popover>
+                    </div>
+
+                    <div className="mt-4 h-[75px] w-full rounded-xl bg-gray-200">
+                        <div className="grid w-full max-w-sm items-center gap-1.5 p-4">
+                            <Label htmlFor="spent">Total Spent</Label>
+                            <p id="spent" className="font-bold text-xl">
+                                ${(quantity * stockPrice).toFixed(2)}
```
Comment (DavidLHW) [inline components/AddTransactionModal.tsx:296]: can use the [`formatUsd(quantity * stockPrice)` funtion](https://github.com/is442oop/portfolio-analyzer-web-ui/blob/main/utils/functions.ts#L1) like:

```ts
import { formatUsd } from "@/utils/functions";

const quantity = 10.0
const stockPrice = 25000.0;

console.log(formatUsd(quantity * stockPrice));  // prints '$250,000.00'
```

example use case: [@/components/DashboardHeader.tsx:51](https://github.com/is442oop/portfolio-analyzer-web-ui/blob/main/components/DashboardHeader.tsx#L51)

**Review by DavidLHW — COMMENTED** [review]
thanks for the PR!

left some comments, other than those, mostly seems good :)

File: components/AddTransactionModal.tsx (hunk 1 above)
Comment (lionelgwk) [reply components/AddTransactionModal.tsx]: thanks for the comments! have hidden the key and regenerated it! good catch!

File: components/AddTransactionModal.tsx (hunk 2 above)
Comment (lionelgwk) [reply components/AddTransactionModal.tsx]: thanks for pointing that out, made use of that so the display for the amount is more unified and standard!

**Review by jeromeandrewong — APPROVED** [review]
LGTM Thanks for the PR

**Pull request merged**